transform.go

```diff
@@ -117,6 +117,42 @@ func Zip[T1, T2 any](seq1 Seq[T1], seq2 Seq[T2]) Seq[Zipped[T1, T2]] {
 	}
 }
 
+// Zip returns a new Seq that yields the values of seq1 and seq2
+// simultaneously.
+func Zip[T1, T2 any](seq1 Seq[T1], seq2 Seq[T2]) Seq[Zipped[T1, T2]] {
+	return func(body func(Zipped[T1, T2]) bool) {
```

Comment:
I'm not thrilled with `body` as the name of the yield function. Why not just `yield` like everywhere else?